src/Services/Authentication/IClaimsSerializer.cs

```diff
@@ -0,0 +1,11 @@
+using System.IO;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication;
+
+public interface IClaimsSerializer
+{
+    ClaimsPrincipal? Deserialize(Stream? responseContent);
+    void Serialize(ClaimsPrincipal? claimsPrincipal, Stream? outputStream);
```

Comment:
Why the nullable? Don't we want to only handle this if we are actually given a stream/principal? I understand the deserialize returning null potentially, but the others should probably be non-nullable

Reply:
Good catch. I'll make the parameters non-nullable.